microservices/services/contest-service/src/main/java/org/xcolab/service/proposal/domain/proposal/ProposalDaoImpl.java

```diff
@@ -345,4 +363,109 @@ public Integer getProposalMaterializedPoints(Long proposalId) {
+                .and(ACTIVITY_ENTRY.CLASS_PRIMARY_KEY.in(orphanProposals))
+                .execute();
+        // Select proposal groups.
+        List<Long> groups = ctx.select(PROPOSAL.GROUP_ID).from(PROPOSAL)
+                .where(PROPOSAL.PROPOSAL_ID.in(orphanProposals))
+                .fetchInto(Long.class);
+        // Delete proposal group members
+        ctx.deleteFrom(USERS_GROUPS)
+                .where(USERS_GROUPS.GROUP_ID.in(groups))
+                .execute();
+        // Delete group membership requests
+        ctx.deleteFrom(MEMBERSHIP_REQUEST)
+                .where(MEMBERSHIP_REQUEST.GROUP_ID.in(groups))
+                .execute();
+        // Delete proposal groups
+        ctx.deleteFrom(GROUP_)
+                .where(GROUP_.GROUP_ID.in(groups))
+                .execute();
+        // Delete proposal supports
+        ctx.deleteFrom(PROPOSAL_SUPPORTER)
+                .where(PROPOSAL_SUPPORTER.PROPOSAL_ID.in(orphanProposals))
+                .execute();
+        // Delete proposal votes
+        ctx.deleteFrom(PROPOSAL_VOTE)
+                .where(PROPOSAL_VOTE.PROPOSAL_ID.in(orphanProposals));
+        // Delete contest phase attributes
+        ctx.deleteFrom(PROPOSAL_CONTEST_PHASE_ATTRIBUTE)
+                .where(PROPOSAL_CONTEST_PHASE_ATTRIBUTE.PROPOSAL_ID.in(orphanProposals))
+                .execute();
+        // Delete proposal ratings
+        ctx.deleteFrom(PROPOSAL_RATING)
+                .where(PROPOSAL_RATING.PROPOSAL_ID.in(orphanProposals))
+                .execute();
+        // Delete proposal references
+        ctx.deleteFrom(PROPOSAL_REFERENCE)
+                .where(PROPOSAL_REFERENCE.PROPOSAL_ID.in(orphanProposals))
+                .or(PROPOSAL_REFERENCE.SUB_PROPOSAL_ID.in(orphanProposals))
+                .execute();
+        // Delete proposal move history entries
+        ctx.deleteFrom(PROPOSAL_MOVE_HISTORY)
+                .where(PROPOSAL_MOVE_HISTORY.SOURCE_PROPOSAL_ID.in(orphanProposals))
+                .or(PROPOSAL_MOVE_HISTORY.TARGET_PROPOSAL_ID.in(orphanProposals))
+                .execute();
+        // Delete points distribution configuration
+        ctx.deleteFrom(POINTS_DISTRIBUTION_CONFIGURATION)
+                .where(POINTS_DISTRIBUTION_CONFIGURATION.PROPOSAL_ID.in(orphanProposals))
+                .or(POINTS_DISTRIBUTION_CONFIGURATION.TARGET_SUB_PROPOSAL_ID.in(orphanProposals))
+                .execute();
+        ctx.deleteFrom(POINTS)
+                .where(POINTS.PROPOSAL_ID.in(orphanProposals))
+                .or(POINTS.ORIGINATING_PROPOSAL_ID.in(orphanProposals))
+                .execute();
+        // Delete orphaned proposals
+        ctx.deleteFrom(PROPOSAL)
+                .whereNotExists(ctx.select(val(1))
+                        .from(PROPOSAL_2_PHASE)
+                        .where(PROPOSAL_2_PHASE.PROPOSAL_ID.eq(PROPOSAL.PROPOSAL_ID)))
```

Comment:
You can just re-use the `PROPOSAL_ID.in(orphanProposals)` condition here. In addition to making absolutely sure that we are really deleting the same proposals that we retrieved earlier, this also reduced duplication (imagine if someone changes the condition at the beginning of the method, but doesn't know the condition is repeated here).

Reply:
Done.